COM Helpers and Various Constants

excaliburHisSheath opened the pull request:
- Adds the RIDL declarations for IMMDevice, IMMDeviceEnumerator, IAudioClient, and IAudioRenderClient.
- Adds `SetPriorityClass()` to kernel32-sys.
- Adds `timeBeginPeriod()` and `timeEndPeriod()` to winmm-sys.
- Adds various constants related to core audio, timers, and thread priorities.

I left the RIDL declarations in the main winapi lib file, but if you want them each in a separate file like you did with unknownbase.rs then I can move them over.

retep998 commented:
As issue #17 states, everything is being split out of `lib.rs` into separate files for each header.

retep998 commented on src/lib.rs:
```diff
@@ -1234,87 +1290,69 @@ pub const IID_IMMDeviceEnumerator: IID = GUID {
-    pub QueryInterface: unsafe extern "system" fn(
-        This: *mut IMMDevice,
+RIDL!(
+interface IMMDevice(IMMDeviceVtbl) {
```
Could you change the COM definitions to also inherit from their base COM classes, like the way I did with `AsyncIUnknown`?

excaliburHisSheath replied:
No problem, it's taken care of.

excaliburHisSheath commented:
I moved all the things I added or changed into the appropriate modules.

retep998 commented on src/audioclient.rs:
```diff
@@ -2,6 +2,8 @@
 // Licensed under the MIT License <LICENSE.md>
 //! this ALWAYS GENERATED file contains the definitions for the interfaces
 //1627
+use unknwnbase::*;
```
I'd prefer not to do this.

excaliburHisSheath replied:
If I try to do `::IUnknown` the RIDL macro doesn't compile, so I need to import the type manually. Would it be better to change the macro so that it supports absolute paths?

retep998 replied:
It would be better to fix the macro.

excaliburHisSheath replied:
This should be fixed now.

retep998 commented:
Thanks, could you rebase everything and maybe ![squash](http://vignette3.wikia.nocookie.net/farmville2/images/2/23/Squash.png/revision/latest) it a bit?

excaliburHisSheath commented:
Rebased and squashed. Let me know if it looks like I did anything wrong, I don't normally rebase/squash in my projects so I wasn't familiar with the process.
